### PR TITLE
test: Add stub client for testing easily.

### PR DESCRIPTION
**Pull request by morefreeze**
It is preparing for testing region split. If it's ok we do need add more test cases.
PTAL

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -28,8 +28,16 @@ import (
...
+// Client is a client that sends RPC.
+// It should not be used after calling Close().
+type Client interface {
+	Close() error
+	SendKVReq(req *kvrpcpb.Request) (*kvrpcpb.Response, error)
+	SendCopReq(req *coprocessor.Request) (*coprocessor.Response, error)
+}
+
+// pbClient connect kvserver to send Request by TCP.
+type pbClient struct {
```
Comment (shenli) [inline client.go]: "client" is a better name?

File: kv_test.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	c.Assert(region, NotNil)
+	expReq := makeGetReq(region, key, txn.tid)
+	expResp := makeGetResp(val)
+	s.client.push(expReq, expResp)
+	value, err := txn.Get([]byte(key))
+	c.Assert(err, IsNil)
+	c.Assert(value, BytesEquals, []byte(val))
+}
+
+// stubClient it used for simulating sending Request to TiKV and receiving Response.
```
Comment (shenli) [inline kv_test.go]: it -> is

File: client.go (hunk 1 above)
Comment (morefreeze) [inline client.go]: I think "client" is a very common name, and it is confused with "clients" in `kv.go`(e.g.: `tikvStore.getClient`). So I add a prefix to make a difference.

**Comment by morefreeze** [comment]
PTAL

File: client.go [hunk 3]
```diff
@@ -28,8 +28,16 @@ import (
 	"gopkg.in/fatih/pool.v2"
 )
 
-// Client connect kvserver to send Request by TCP.
-type Client struct {
+// Client is a client that sends RPC.
+// It should not be used after calling Close().
+type Client interface {
+	Close() error
+	SendKVReq(req *kvrpcpb.Request) (*kvrpcpb.Response, error)
```
Comment (shenli) [inline client.go]: Public interface should have comment.

File: client.go (hunk 1 above)
Comment (shenli) [inline client.go]: stubClient also accept pb args.
I think rpcClient may be a better name? And RPCClient may be a better name for Client interface.

**Comment by shenli** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Pull request merged**